### PR TITLE
Clean up TODO in hack/generate-code

### PR DESCRIPTION
**Pull request by ialidzhikov**
**What this PR does / why we need it**:
- Clean up TODO in hack/generate-code
- Fix some golangci-lint findings
- Remove protobuf tags as CRDs cannot support protocol buffers - ref https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/#advanced-features-and-flexibility

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement operator
NONE
```

### CODE REVIEW

**Comment by hardikdr** [comment]
thanks for the PR @ialidzhikov , this is a much-needed clean-up,  I'll be taking a look soon.

File: pkg/driver/driver_azure.go [hunk 1]
```diff
@@ -786,23 +773,6 @@ func (clients *azureDriverClients) waitForDataDiskDetachment(ctx context.Context
 	return nil
 }
 
-func (clients *azureDriverClients) powerOffVM(ctx context.Context, resourceGroupName string, vmName string) error {
```
Comment (hardikdr) [inline pkg/driver/driver_azure.go:789]: @dkistner Do we plan to use this in near future, or should it better be removed ?

File: hack/generate-code [hunk 2]
```diff
@@ -19,24 +19,8 @@ rm -f $GOPATH/bin/*-gen
...
-# Go modules are not able to properly vendor k8s.io/code-generator@kubernetes-1.12.1 (build with godep)
-# to include also generate-internal-groups.sh scripts under vendor/.
-# However this is fixed with kubernetes-1.15.0 (k8s.io/code-generator@kubernetes-1.15.0 is opted in go modules).
-# The workaround for now is to have a kubernetes-1.12.1 copy of generate-internal-groups.sh under hack/code-generator and
-# to copy them to vendor/k8s.io/code-generator/ for the generation.
-# And respectively clean up them after execution.
-# The workaround should be removed adopting kubernetes-1.15.0.
-# Similar thing is also done in https://github.com/heptio/contour/pull/1010.
-
-cp "${CURRENT_DIR}"/code-generator/generate-internal-groups.sh "${PROJECT_ROOT}"/vendor/k8s.io/code-generator/
```
[the remark this answers is not in the document]
Comment (ialidzhikov) [reply hack/generate-code:31]: As described in the comments above, it was not possible to have `generate-internal-groups.sh` located under vendor as k8s.io/code-generator < kubernetes-1.15.0 is not switched to go modules. As the k/k dependency are already updated to v1.16.0, we can remove this workaround and have the `generate-internal-groups.sh` under `vendor/k8s.io/code-generator/`.

File: pkg/driver/driver_alicloud.go [hunk 3]
```diff
@@ -132,21 +132,6 @@ func (c *AlicloudDriver) Delete(machineID string) error {
 	return err
 }
 
-func (c *AlicloudDriver) stopInstance(client *ecs.Client, machineID string) error {
```
[the remark this answers is not in the document]
Comment (ialidzhikov) [reply pkg/driver/driver_alicloud.go:135]: We have VCS and a source code won't be lost. No need to keep unused things in the master branch, if a something is required we can always check the revisions back.

[the remark this answers is not in the document]
Comment (hardikdr) [reply pkg/driver/driver_alicloud.go:135]: Agreed, felt the same later.

File: pkg/driver/driver_azure.go (hunk 1 above)
Comment (hardikdr) [reply pkg/driver/driver_azure.go:789]: We can always re-introduce it if needed.

**Comment by hardikdr** [comment]
I shall merge it by the end of the day unless @prashanth26 or @ggaurav10 have any concerns or might want to take a look.

**Pull request merged**